<!DOCTYPE html>
<html>
  <head>
  <meta charset=utf-8 />
  <title>  MCB111 Mathematics in Biology </title>
  <link rel="stylesheet" href="/Harvard-MCB111-2024-Fall/css/style.css" />
</head>

<script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>

<body>
  <div id="main">
 
    <header>
      <h3> MCB111: Mathematics in Biology (Fall 2024) </h3>
    </header>
 
    <nav role="navigation"> 
      <a href="/Harvard-MCB111-2024-Fall">Home</a>  | <a href="/Harvard-MCB111-2024-Fall/schedule.html">Schedule</a> | <a href="https://canvas.harvard.edu/courses/139673/">Canvas</a> | <a href="https://piazza.com/harvard/fall2024/mcb111">Piazza</a> | <a href="/Harvard-MCB111-2024-Fall/downloads/MCB111-syllabus.pdf">Syllabus [PDF]</a> | <a href="/Harvard-MCB111-2024-Fall/downloads/StudentHours_2024.pdf">Student hours schedule [PDF]</a> <br/>
      
      Lectures:
      <a href="/Harvard-MCB111-2024-Fall/w00/w00-lecture.html"> w00 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w01/w01-lecture.html"> w01 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w02/w02-lecture.html"> w02 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w03/w03-lecture.html"> w03 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w04/w04-lecture.html"> w04 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w05/w05-lecture.html"> w05 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w06/w06-lecture.html"> w06 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w07/w07-lecture.html"> w07 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w08/w08-lecture.html"> w08 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w09/w09-lecture.html"> w09 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w10/w10-lecture.html"> w10 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w11/w11-lecture.html"> w11 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w12/w12-lecture.html"> w12 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w13/w13-lecture.html"> w13 </a> |
      <br/>
      
      inclass-notes:
      <a href="/Harvard-MCB111-2024-Fall/w02/w02-notes-lecture1.pdf"> w02-l1 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w02/w02-notes-lecture2.pdf"> w02-l2 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w03/w03-notes-lecture1.pdf"> w03-l1 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w04/w04-notes-l1.pdf"> w04-l1 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w05/w05-notes-lectures.pdf"> w05 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w06/w06-notes-l1.pdf"> w06-l1 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w06/w06-notes-l2.pdf"> w06-l2 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w07/w07-notes-l1.pdf"> w07-l1 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w07/w07-notes-l2.pdf"> w07-l2 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w08/w08-notes-l1.pdf"> w08-l1 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w08/w08-notes-l2.pdf"> w08-l2 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w09/w09-notes-l1.pdf"> w09-l1 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w09/w09-notes-l2.pdf"> w09-l2 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w10/w10-notes-l1.pdf"> w10-l1 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w10/w10-notes-l2.pdf"> w10-l2 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w10/w10-notes-l3.pdf"> w10-l3 </a> |
       <br/>
      
      inclass-code: 
      <a href="/Harvard-MCB111-2024-Fall/w00/w00-inclass.html"> w00 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w01/w01-sections_er_2024_complete.html"> w01 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w02/w02-lecture1.html"> w02-l1 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w02/w02-lecture2.html"> w02-l2 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w03/w03-lecture-code.ipynb"> w03 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w04/w04-lecture-code.ipynb"> w04 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w13/w13-code.ipynb"> w13 </a> |
      <br/>
      
      Sections: 
      <a href="/Harvard-MCB111-2024-Fall/w01/w01-sections.html"> w01 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w02/w02-sections.html"> w02 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w03/w03-sections_NH_2024.ipynb"> w03 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w04/w04-sections_2024.ipynb"> w04 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w05/w05-sections_2024.ipynb"> w05 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w08/w08-sections_2024.ipynb"> w08 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w09/w09-sections_2024-ER.ipynb"> w09 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w10/w10-sections_2024-NH.ipynb"> w10 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w11/w11-sections_2024_er.ipynb"> w11 </a> |
     <br/>
      
      
      Homeworks:
      <a href="/Harvard-MCB111-2024-Fall/w00/w00-homework.html"> w00 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w01/w01-homework.html"> w01 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w02/w02-homework.html"> w02 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w03/w03-homework.html"> w03 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w04/w04-homework.html"> w04 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w05/w05-homework.html"> w05 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w06/w06-homework.html"> w06 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w07/w07-homework.html"> w07 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w08/w08-homework.html"> w08 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w09/w09-homework.html"> w09 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w10/w10-homework.html"> w10 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w11/w11-homework.html"> w11 </a> |
      <a href="/Harvard-MCB111-2024-Fall/w13/w13-homework.html"> w13 </a> |
     <br/>
      
      Hints: 
      <a href="/Harvard-MCB111-2024-Fall/w00/w00-hints.html"> w00 </a> |
      <br/>
      

      Final:
      <br/>
 
     </nav>
    
    <br/>
 
    <ul id="markdown-toc">
  <li><a href="#the-length-of-a-unique-sequence-in-a-genome" id="markdown-toc-the-length-of-a-unique-sequence-in-a-genome">The length of a unique sequence in a genome</a>    <ul>
      <li><a href="#how-do-we-address-this-question" id="markdown-toc-how-do-we-address-this-question">How do we address this question?</a></li>
      <li><a href="#lets-check-the-result-by-simulation" id="markdown-toc-lets-check-the-result-by-simulation">Let’s check the result by simulation</a></li>
    </ul>
  </li>
</ul>

<h1 class="no_toc" id="week-01">week 01:</h1>

<h2 id="the-length-of-a-unique-sequence-in-a-genome">The length of a unique sequence in a genome</h2>

<p>In a genome of length \(L\), what is the shortest sequence (motif) that we
expect to appear only once?</p>

<p>An alternative way of asking the same question: how long has to be a
motif for it to match only one location in the genome?</p>

<p>Eukaryotic genomes have on the order of \(L=10^{6}-10^{10}\)
nucleotides.  We assume that the length of the motif is \(l\).  Let’s
also assume a that the base composition of the genome is such that all
four bases (A, C, G, T) are equally likely. Then, a motif of length
\(l=1\) (for instance “A”), will match \(1/4\) of the positions in the
genome. If we take a motif of length \(l=2\), for instance “AA” it
will match fewer than \(1/4\), but not much better.</p>

<h3 id="how-do-we-address-this-question">How do we address this question?</h3>

<div id="figcontainer">
     <div id="figure">
     	  <img src="unique.jpg" style="width:300px;" /> <br />
	  
	  Figure 2. The ignorance about the location in the genome is
  	  determined by the length of the genome L as log_2(L), and it
  	  doesn't depend on the length of the motif l (as long as l&lt;&lt;L
  	  and we can ignore edges). The total number of possible
  	  motifs is 4^l, and the information content of each one of
  	  them is log_2(4^l)= 2l (assuming all residues are equally
  	  likely).  In a situation where the segment size is small,
  	  the total number of possible segments is small (for
  	  instance, for l=2 there are 16 possible segments), thus we
  	  expect that none of the segments is going to be unique in the
  	  genome.  As the segment size increases, the number of
  	  possible segments does too, so does the information content
  	  of each one of them (for instance, for l=16 there are over 4
  	  billion possible segments) then it becomes more likely that
  	  a given segment is unique in the genome.
	</div>
</div>

<ul>
  <li>
    <p>On the one hand, we have the information of where the sequence is
going to land in the genome (the genomic position), if we assume
all \(L\) positions are equally likely we have, then a given motif
can fall starting at any of the \(L\) positions in the genome. (The
exact nubmer of starting positions for a motif of length \(l\) is
\(L-l\), but it we assume \(l&lt;&lt; L\) then \(L-l\approx L\).)</p>

    <p>Then, the information in the genome position of the motif is</p>

\[\log_2 L\]
  </li>
  <li>
    <p>On the other hand, we have the information in the motif. That is, using a
combination of A, C, G, T’s how many different motifs can we make that
   have length \(l\)</p>

    <p>If we assume that each position is independent from each other, so
their entropies add up, and each position can be one of the four
residues, so the number of possibilities for one position is 4, and</p>

    <p>Information content on  a motif with l positions is</p>

\[I(l) =  l\ \log_2(4) = 2\ l\]

    <p>Since the probability of each nucleic acid is</p>

\[p_A = p_C = p_G = p_T = \frac{1}{4}\]

    <p>we can re-written the information of the motif as</p>

\[I(l) =  2\ l = l\ \log_2(4) = l\ \log_2\frac{1}{1/4},\]

    <p>that is, the sum of the information content of \(l\) sites each with probability \(\frac{1}{4}\).</p>
  </li>
</ul>

<p>For the motif to be unique, we want the information on the motif composition to be at least as large as that of the possible location of the motif, that is
We want to have the condition  (see Figure 2)</p>

\[\mbox{Information( motif)}\quad &gt;=\quad \mbox{Information(genome position)}\]

<p>or</p>

\[H(l) &gt;= \log_2(L)\]

<p>that we require that the length of the motif \(l\) is at least</p>

\[l = \frac{1}{2}\ log_2(L)\]

<h3 id="lets-check-the-result-by-simulation">Let’s check the result by simulation</h3>

<p>Let’s look at some examples. Assume a genome of length \(L = 10^6\).</p>

<ul>
  <li>
    <p>The theoretical results</p>

    <p>The  ``location information’’ is
	\(\log_2{10^6}=19.93\, \mbox{bits},\)</p>
  </li>
</ul>

<p>and the entropy for different motifs of different lengths are</p>

<table>
  <thead>
    <tr>
      <th>length</th>
      <th>l=4</th>
      <th style="text-align: center">l=4</th>
      <th style="text-align: center">l=6</th>
      <th style="text-align: center">l=6</th>
      <th style="text-align: center">l = 10</th>
      <th style="text-align: center">l= 10</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>motif</td>
      <td>AAAA</td>
      <td style="text-align: center">CCCC</td>
      <td style="text-align: center">AAAAAA</td>
      <td style="text-align: center">CCCCCC</td>
      <td style="text-align: center">AAAAAAAAAA</td>
      <td style="text-align: center">CCCCCCCCCC</td>
    </tr>
  </tbody>
  <tbody>
    <tr>
      <td>I(motif)</td>
      <td>8.0</td>
      <td style="text-align: center">8.0</td>
      <td style="text-align: center">12.0</td>
      <td style="text-align: center">12.0</td>
      <td style="text-align: center"><strong>20.0</strong></td>
      <td style="text-align: center"><strong>20.0</strong></td>
    </tr>
  </tbody>
</table>

<p>In bold, situations in which the motif’s information is larger than the
genome position information, and we should expect the motif to be
unique in the genome.</p>

<p>For a genome where all nucleotides are distributed equally and for a
genome of length \(10^6\), sequences of length \(10\) or larger will
already be unique.</p>

<ul>
  <li>A simulation</li>
</ul>

<p>Here, I simulate random genomes of length \(10^6\), and we calculate
  how many times the different motifs were found. By generating
  several different genomes, we can also estimate the mean and
  variance.</p>

<p>Here are actual results using 20 random genomes, and assuming that all nucleotides have the same probability \(1/4\),</p>

<table>
  <thead>
    <tr>
      <th>lenght</th>
      <th>l=4</th>
      <th style="text-align: center">l=4</th>
      <th style="text-align: center">l=6</th>
      <th style="text-align: center">l=6</th>
      <th style="text-align: center">l = 10</th>
      <th style="text-align: center">l= 10</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>motif</td>
      <td>AAAA</td>
      <td style="text-align: center">CCCC</td>
      <td style="text-align: center">AAAAAA</td>
      <td style="text-align: center">CCCCCC</td>
      <td style="text-align: center">AAAAAAAAAA</td>
      <td style="text-align: center">CCCCCCCCCC</td>
    </tr>
  </tbody>
  <tbody>
    <tr>
      <td>occurence</td>
      <td>3,894\(\pm\)88</td>
      <td style="text-align: center">3,904\(\pm\)93</td>
      <td style="text-align: center">240\(\pm\)13</td>
      <td style="text-align: center">247\(\pm\)25</td>
      <td style="text-align: center">1.2\(\pm\)1.2</td>
      <td style="text-align: center">0.9\(\pm\)1.2</td>
    </tr>
  </tbody>
</table>

<p>As a rule of thumb, for a genome with residues uniformly distributed
(\(p_A=p_C=p_G=p_T=0.25\)), then the information content of a nucleotide
is 2 bits (\(\log_2\frac{1}{0.25}\)) and the length of a unique motif is
\begin{equation}
l \geq \frac{\log_2(L)}{2}.
\end{equation}</p>

<p>Some examples,</p>

<table>
  <thead>
    <tr>
      <th>genome</th>
      <th>length \(L\)</th>
      <th>genome inf content</th>
      <th>unique motif length \(l\)</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>human</td>
      <td>\(4\times 10^9\)</td>
      <td>32 bits</td>
      <td>16</td>
    </tr>
    <tr>
      <td>worm</td>
      <td>\(100\times 10^6\)</td>
      <td>27 bits</td>
      <td>13</td>
    </tr>
    <tr>
      <td>E.coli</td>
      <td>\(6\times 10^6\)</td>
      <td>22 bits</td>
      <td>11</td>
    </tr>
  </tbody>
</table>


 
    <footer>
      <hr>
      <p>
        <a href="http://rivaslab.org">Elena Rivas</a> | Harvard University
    </footer>
 
  </div>
</body>
</html>
